bbc/simorgh
# WSTEAM1-1361: Add IG/YT social embed support on Live Page

Resolves JIRA **WSTEAM1-1361**

Overall changes
======
_A very high-level summary of easily-reproducible changes that can be understood by non-devs, and why these changes where made._

Code changes
======

- Add support for rendering Youtube and Instagram embeds.

Testing
======
1. In FABL, check out branch `WSTEAM1-1361-spike-ig-yt-embeds-live-page` of the BFF.
2. Run BFF locally.
3. In Simorgh, set to BFF_PATH environment variable to `http://localhost:3210/module/simorgh-bff`
4. Navigate to the `ws-nextjs-app` directory and run `yarn dev`.
5. Visit http://localhost:7081/mundo/live/c5w1wlz68l4t?renderer_env=test
6. Click the consent banners to reveal the embeds.
7. View the Instagram and Youtube embeds. 

a11y checks:

- [x] Skip links work with a keyboard
- [x] The text in the skip links have the correct context e.g. skip instagram post
- [x] The text in the skip link end message has the correct context e.g. end of instagram post
- [x] If there is more than 1 post on the page with the same type of embed, that the skip link is labelled with a number e.g. skip instagram post 2 - THIS IS DONE NORMALLY ( [example w multiple twitter embeds in a post here](https://simorgh.belfrage-preview.test.api.bbc.com/pidgin/live/c7p765ynk9qt))



Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

#### Instagram consent banner
![Screenshot 2024-10-15 at 18 07 21](https://github.com/user-attachments/assets/bfc386ae-b880-47e2-9c8d-52ab2eebbb8f)

#### Instagram embed
![Screenshot 2024-10-16 at 17 39 40](https://github.com/user-attachments/assets/4f1cb763-badd-4112-a319-0bddf856e33a)

#### Youtube consent banner
![Screenshot 2024-10-15 at 18 07 52](https://github.com/user-attachments/assets/f04f86f4-45d0-49f3-b4b5-21339912f095)

#### Youtube embed
![Screenshot 2024-10-15 at 18 08 21](https://github.com/user-attachments/assets/12b4d754-ca63-4b58-b704-6fb6b20b13d7)
